pkg/resources/core/v1/namespace/psalabels.go

```diff
@@ -60,13 +68,27 @@ func (p *psaLabelAdmitter) Admit(request *admission.Request) (*admissionv1.Admis
 		extras[k] = v1.ExtraValue(v)
 	}
 
+	var projectNamespace, projectName string
+	// here we are filling the variables above with the projectId,
+	// so that if we are not able to get them,
+	// the SAR request will be done in any case.
+	if ns.Annotations[projectID] != "" {
+		projectInfo := strings.Split(ns.Annotations[projectID], ":")
```

Comment:
Not necessary, but I generally prefer to use `strings.Cut` for this operation:
```suggestion
		projectNamespace, projectName, found := strings.Cut(ns.Annotations[projectID], ":")
		if found {
		...
		}
```